src/Stan/Analysis/Pretty.hs

```diff
@@ -81,6 +81,6 @@ showByFile (file, o :| obs) = unlines
     b = formatWith [bold]
 
 groupObservationsByFile
-    :: [Observation]
+    :: Observations
     -> HashMap FilePath (NonEmpty Observation)
```

Comment:
Do you think it makes sense to change the type here to `HashMap FilePath Observations` so we can tell the total number of observations in a group faster?